### PR TITLE
Send redirect url for oauth2

### PR DESCRIPTION
**Pull request by furkanayhan**
I've also fixed some indentations in the actions file
There are two separate commit, so you can easily check the changes.

Why I need redirect url to be sent?
That's because servers return 401 unless sending it.

### CODE REVIEW

File: src/core/plugins/auth/actions.js [hunk 1]
```diff
@@ -104,17 +109,17 @@ export const authorizeApplication = ( auth ) => ( { authActions } ) => {
...
-
- return authActions.authorizeRequest({body: buildFormData(form), name, url: schema.get("tokenUrl"), auth})
+export const authorizeAccessCode = ( { auth, redirectUrl } ) => ( { authActions } ) => {
+  let { schema, name, clientId, clientSecret } = auth
+  let form = {
+    grant_type: "authorization_code",
+    code: auth.code,
+    client_id: clientId,
+    client_secret: clientSecret,
+    redirect_uri: redirectUrl
```
Comment (furkanayhan) [inline src/core/plugins/auth/actions.js]: the only related change for this file is this.

**Comment by furkanayhan** [comment]
also fixes #3034

**Comment by furkanayhan** [comment]
@webron @bodnia is there something wrong with this PR?

**Comment by furkanayhan** [comment]
yes this broke my fix: https://github.com/swagger-api/swagger-ui/commit/8aebea34c4ee1f063a8a5c6800f0d8643751eff3

**Comment by webron** [comment]
@bodnia can you please review the PR?

File: src/core/oauth2-authorize.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ export default function authorize ( { auth, authActions, errActions, configs, au
   }
 
   url += "&redirect_uri=" + encodeURIComponent(redirectUrl)
-      + "&realm=" + encodeURIComponent(authConfigs.realm);
+      + "&realm=" + encodeURIComponent(authConfigs.realm)
```
Comment (furkanayhan) [inline src/core/oauth2-authorize.js]: this change fixes: https://github.com/swagger-api/swagger-ui/commit/8aebea34c4ee1f063a8a5c6800f0d8643751eff3

File: src/core/plugins/auth/actions.js (hunk 1 above)
Comment (furkanayhan) [inline src/core/plugins/auth/actions.js]: other changes are just indentation fixes

**Comment by furkanayhan** [comment]
I've rebased and sent again

**Comment by rstraszewski** [comment]
redirect_uri is REQUIRED, if the "redirect_uri" parameter was included in the authorization request. This is token request, so redirect_uri is only for validation purposes, if I understand correctly. Response from token request seems always to be json with access_token, not redirect. https://security.stackexchange.com/questions/44214/what-is-the-purpose-of-oauth-2-0-redirect-uri-checking

**Comment by bodnia** [comment]
@rstraszewski  could you please resolve conflicts and I'll merge

**Comment by rstraszewski** [comment]
Sorry, not my pull request, just interested in this change, so added my two cents :)

**Comment by bodnia** [comment]
@rstraszewski sorry about this :)

@furkanayhan could you please update branch to latest master

**Comment by furkanayhan** [comment]
@rstraszewski :)
@bodnia this commit broke my pr https://github.com/swagger-api/swagger-ui/commit/960a99a5eaf0a4831da0e44176026d8a92b2bb19, then I ignored my change on that file.
fyi

**Comment by viswatejan** [comment]
I am getting **Cannot read property 'schema' of undefined** on oauth2-redirect.html page with **accessCode** oauth flow.

I tested on master branch dist folder. Any update on this?

**Comment by rstraszewski** [comment]
I had the same problem, but I tried html files from directory dev-helpers and everything worked fine. I guess, there are some additional steps in oauth configuration compared to previous versions.

**Comment by viswatejan** [comment]
3.0.14 is working perfectly.

**Pull request merged**